aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py

```diff
@@ -219,14 +239,82 @@ async def filter_constraints(
         if not applicable:
             continue
 
-        # TODO: create new credential with selective disclosure
-        if constraints.limit_disclosure or predicate:
-            raise PresentationExchError("Not yet implemented - createNewCredential")
+        if constraints.limit_disclosure:
+            credential_dict = json.loads(credential.cred_value)
+            new_credential_dict = reveal_doc(
+                credential_dict=credential_dict, constraints=constraints
+            )
 
+            # Using custom_document_loader for testing
+            signed_new_credential_dict = await derive_credential(
+                credential=credential_dict,
+                reveal_document=new_credential_dict,
+                suite=suite,
+                document_loader=document_loader,
+            )
+            credential = VCRecord.deserialize_jsonld_cred(
+                json.dumps(signed_new_credential_dict)
+            )
         result.append(credential)
     return result
 
 
+def reveal_doc(credential_dict: dict, constraints: Constraints):
+    new_credential_dict = {}
+    new_credential_dict["@context"] = credential_dict.get("@context")
+    new_credential_dict["type"] = credential_dict.get("type")
+    new_credential_dict["@explicit"] = True
+    new_credential_dict["issuanceDate"] = credential_dict.get("issuanceDate")
+    unflatten_dict = {}
+    for field in constraints._fields:
```

Comment:
What if none of the fields matches? Should that throw an error? Or is that not possible?